### PR TITLE
Issue 463

### PR DESCRIPTION
**Pull request by mchasteen**
Tried to fix most SQL inserts and update issues with strict mode.  Some of these functions probably need to be reworked.  I got most of the functionality working.  I tested it with some Cisco switches and with an APC UPS.  There still is an issue with sensors.

### CODE REVIEW

File: includes/dbFacile.php [hunk 1]
```diff
@@ -24,10 +24,10 @@
 function dbQuery($sql, $parameters = array()) {
 	global $fullSql, $debug;
 	$fullSql = dbMakeQuery($sql, $parameters);
-        if($debug) { 
-          print Console_Color::convert("\nSQL[%y".$fullSql."%n] ");
-          #echo("\nSQL[".$fullSql."] "); 
-        }
+        //if($debug) {
```
Comment (laf) [inline includes/dbFacile.php]: This needs to stay in really, it's for running poller and discovery in debug mode.

File: includes/dbFacile.php [hunk 2]
```diff
@@ -114,9 +118,13 @@ function dbUpdate($data, $table, $where = null, $parameters = array()) {
 
 	// need field name and placeholder value
 	// but how merge these field placeholders with actual $parameters array for the WHERE clause
-	$sql = 'UPDATE `' . $table . '` set ';
+	$sql = 'UPDATE `' . $table . '` SET ';
 	foreach($data as $key => $value) {
-                $sql .= "`".$key."` ". '=:' . $key . ',';
```
Comment (laf) [inline includes/dbFacile.php]: Think we should fix the query rather than alter dbFacile. Do you know where the query was being called from that is broke?

File: includes/discovery/entity-physical.inc.php [hunk 3]
```diff
@@ -30,10 +30,16 @@
...
-    if (isset($entity_array['$entPhysicalIndex']['0']['entAliasMappingIdentifier'])) { $ifIndex = $entity_array['$entPhysicalIndex']['0']['entAliasMappingIdentifier']; }
+    if (isset($entity_array[$entPhysicalIndex]['0']['entAliasMappingIdentifier'])) {
+      $ifIndex = $entity_array[$entPhysicalIndex]['0']['entAliasMappingIdentifier'];
+    }
 
-    if (!strpos($ifIndex, "fIndex") || $ifIndex == "") { unset($ifIndex);  }
-    list(,$ifIndex) = explode(".", $ifIndex);
+    if (!strpos($ifIndex, "fIndex") || $ifIndex == "") {
+      unset($ifIndex);
+    } else {
```
Comment (laf) [inline includes/discovery/entity-physical.inc.php]: This wasn't an else before and was done after the if statement, is this correct?

**Comment by laf** [comment]
The comments aren't necessarily in the best place for you to view them so you probably be better to view them via Files Changed.

I'm going to test this on my install now.

File: includes/dbFacile.php [hunk 4]
```diff
@@ -35,6 +35,10 @@ function dbQuery($sql, $parameters = array()) {
 	*/
 
 	$result = mysql_query($fullSql); // sets $this->result
+       	if (!empty(mysql_error())) {
```
Comment (laf) [inline includes/dbFacile.php]: This needs to be:

if (mysql_error()) {

Otherwise you get:

Fatal error: Can't use function return value in write context in includes/dbFacile.php on line 38

File: includes/dbFacile.php (hunk 1 above)
Comment (mchasteen) [inline includes/dbFacile.php]: whoops!  I was trying to do debug on just SQL errors.  Might be a good idea to output SQL errors to the log file or a separate log file.

Comment (laf) [inline includes/dbFacile.php]: No worries, if you can remove that for now and we can do another PR with SQL logging.

File: includes/dbFacile.php (hunk 2 above)
Comment (mchasteen) [inline includes/dbFacile.php]: The NOW() value is called by a few functions like at the end of device discovery and poller.

 $sql .= "`".$key."` ". '=:' . $key . ',';
This can be changed back.  I see what it is doing in the dbMakeQuery.  Just looked weird at first.

Comment (laf) [inline includes/dbFacile.php]: Yeah those NOW() calls work ok and dbFacile is a standard library from elsewhere so it's easier if we can just update that with little issue.

File: includes/discovery/entity-physical.inc.php (hunk 3 above)
Comment (mchasteen) [inline includes/discovery/entity-physical.inc.php]: No the else wasn't there.  It's only setting the $ifIndex variable if there is data in it.  Just a different way to do the same thing as list(,$ifIndex)

**Comment by laf** [comment]
I've tested this on my install and it works really well. I'm sure we will see more issues for things that we both don't run ourselves but what you've done is absolutely amazing!!

You will need to rebase against the current master as a couple of changes conflict now but nothing major.

File: includes/dbFacile.php (hunk 2 above)
Comment (mchasteen) [inline includes/dbFacile.php]: I'll switch this back and see what blows up.

File: includes/discovery/entity-physical.inc.php (hunk 3 above)
Comment (laf) [inline includes/discovery/entity-physical.inc.php]: Ok, no worries. More readable now :)

**Comment by mchasteen** [comment]
I'm working on fixing the dbFacile and a few other things.  I usually have been using git for my small projects, so I am still getting used to working with forks and pull requests and keeping things current.

**Comment by laf** [comment]
No problem @mchasteen. Bit of a learning curve but it's powerful :)

**Comment by laf** [comment]
Thanks for updating @mchasteen. Are you able to rebase this branch against the current master as it's got a conflict somewhere so needs fixing manually.

If not, let us know and we can pick out your changes and merge manually.

**Comment by mchasteen** [comment]
Ill try to figure it out!

**Comment by laf** [comment]
Thanks for the update to this @mchasteen.

Git can be a steep learning curve sometimes :) Unfortunately on this one you've also merged a couple of other PRs, one that's already merged in and the fix for alerting which was done in a different way.

If you want some help with this it may be worth jumping on irc or I could create a diff file that you can run against a clean branch and then submit a new PR?

**Comment by laf** [comment]
Thanks for all your help with this @mchasteen, with a couple of small adjustments @f0o has just merged this in :)
